# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/bhavanathiru13-droid/flyrank-AI_1/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane 1: Ranking Signal Analysis

I chose Lane 1 because I want to understand which safe content and search signals are associated with visibility, clicks, engagement, or movement. This lane fits the starter dataset because I can begin with exploratory analysis, grouped summaries, and correlations before considering a model.

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [5]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


The decision is to identify which safe content and search signals are associated with better visibility, clicks, or engagement. A content or SEO team could use these findings to decide which signals are worth reviewing when improving pages. A wrong recommendation could lead the team to spend time and resources on changes that do not improve performance, or to overlook useful signals.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [6]:
import pandas as pd

url = "https://raw.githubusercontent.com/flyrank-bih/flyrank-ml-internship-starter/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print("Rows:", len(df))
print("Columns:", len(df.columns))
print("\nSample numeric summary:")
print(df.select_dtypes(include="number").describe().T.head(10))

Rows: 30000
Columns: 44

Sample numeric summary:
                   count          mean           std   min      25%      50%  \
search_volume    27532.0    158.882391   1518.270825   0.0      0.0     10.0   
competition      27532.0      0.146954      0.285241   0.0      0.0      0.0   
cpc              27532.0      0.485342      2.101560   0.0      0.0      0.0   
word_count       22301.0   3107.760325   1452.382598   8.0   2413.0   2877.0   
char_count       22301.0  20665.277835  10115.344042  40.0  15644.0  19116.0   
impressions_90d  30000.0   5200.366300  16838.019547   1.0     81.0    731.0   
clicks_90d       30000.0     16.097333     75.076958   0.0      0.0      1.0   
pageviews_90d    30000.0     49.942467    152.101430   0.0      2.0      8.0   
sessions_90d     30000.0     37.066633    107.069131   1.0      2.0      7.0   
users_90d        30000.0     35.937700    103.748185   1.0      2.0      7.0   

                      75%        max  
search_volume       20.00   740

The starter dataset contains 30,000 rows and 44 columns. The average number of impressions over 90 days is 5,200.37, while the average number of clicks is 16.10. The average number of pageviews is 9.94. These numbers show that the dataset contains useful visibility and engagement measures for exploring relationships between search signals and performance.

In [7]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This analysis can identify observed relationships and patterns between safe content or search signals and performance measures such as visibility, clicks, and engagement. It can support decisions about which signals are worth reviewing. However, the results are observational and cannot prove that a signal causes better performance. The analysis cannot prove Google ranking algorithm factors or guarantee that changing a page will improve its performance.

In [8]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.